From 90d5f6dd17eb1b3601b81fbdbaa26916b9c6ef6c Mon Sep 17 00:00:00 2001
From: 8dcc <8dcc.git@gmail.com>
Date: Sun, 2 Oct 2022 18:49:20 +0200
Subject: [PATCH] Tags for tagged

---
 tabbed/config.def.h |  7 +++++--
 tabbed/tabbed.c     | 10 ++++++++++
 2 files changed, 15 insertions(+), 2 deletions(-)

diff --git a/tabbed/config.def.h b/tabbed/config.def.h
index c1f0ac2..693996d 100644
--- a/tabbed/config.def.h
+++ b/tabbed/config.def.h
@@ -2,9 +2,9 @@
 
 /* appearance */
 static const char font[]       = "dina:size=9";
-static const char* normbgcolor = "#333333";
+static const char* normbgcolor = "#222222";
 static const char* normfgcolor = "#cccccc";
-static const char* selbgcolor  = "#af2626";
+static const char* selbgcolor  = "#444444";
 static const char* selfgcolor  = "#ffffff";
 static const char* urgbgcolor  = "#111111";
 static const char* urgfgcolor  = "#555555";
@@ -15,6 +15,9 @@ static const int tabwidth      = 200;
 static const Bool foreground   = True;
 static Bool urgentswitch       = False;
 
+static const int tg_width = 3;
+static const char* tg_color = "#af2626";
+
 /*
  * Where to place a new tab when it is opened. When npisrelative is True,
  * then the current position is changed + newposition. If npisrelative
diff --git a/tabbed/tabbed.c b/tabbed/tabbed.c
index eafe28a..d9bad44 100644
--- a/tabbed/tabbed.c
+++ b/tabbed/tabbed.c
@@ -68,6 +68,7 @@ typedef struct {
 	XftColor norm[ColLast];
 	XftColor sel[ColLast];
 	XftColor urg[ColLast];
+    XftColor tag;
 	Drawable drawable;
 	GC gc;
 	struct {
@@ -385,6 +386,14 @@ drawtext(const char *text, XftColor col[ColLast])
 
 	XSetForeground(dpy, dc.gc, col[ColBG].pixel);
 	XFillRectangles(dpy, dc.drawable, dc.gc, &r, 1);
+
+    if (col[ColBG].pixel == dc.sel[ColBG].pixel) {
+        XSetForeground(dpy, dc.gc, dc.tag.pixel);
+        XRectangle tg_rect = { dc.x, dc.y, tg_width, dc.h };
+        XFillRectangles(dpy, dc.drawable, dc.gc, &tg_rect, 1);
+        XSetForeground(dpy, dc.gc, col[ColBG].pixel);   // Reset it
+    }
+
 	if (!text)
 		return;
 
@@ -1036,6 +1045,7 @@ setup(void)
 	dc.sel[ColFG] = getcolor(selfgcolor);
 	dc.urg[ColBG] = getcolor(urgbgcolor);
 	dc.urg[ColFG] = getcolor(urgfgcolor);
+	dc.tag = getcolor(tg_color);
 	dc.drawable = XCreatePixmap(dpy, root, ww, wh,
 	                            DefaultDepth(dpy, screen));
 	dc.gc = XCreateGC(dpy, root, 0, 0);
-- 
2.37.3

